### ISSUE TITLE
Bug detecting use of destructured variable

### ISSUE THREAD

**Issue by tracker1** [eslint/eslint#2086]
I get the following error: `modifiedIndex is defined but never used  no-unused-vars`

From this piece of code...

```
function setService(cache, basePath, node) {
  if (!(cache && basePath && node)) return null;
  var entry = getParts(basePath, node);
  var {service, host, port, modifiedIndex} = entry;
  var svc = cache[service] = cache[service] || [];

  for (var i = 0; i < svc.length; i++) {
    //found match on host/port
    if (svc[i].host === host && svc[i].port === port) {
      if (svc[i].modifiedIndex <= modifiedIndex) {
        svc[i] = entry;
        return false;
      }
      return null;
    }
  }
  svc.push(entry); //new entry
  return true;
}
```

It's used inside the comparison check... I do have ES6 features enabled...

**Comment by nzakas** [comment]
Can you paste the complete output?

Also, what version are you using?

**Comment by xjamundx** [comment]
FYI in ESLint `v0.20.01` I get the following instead:

```js
~/dev/paypal/forklifter (master) $ eslint test.js 

test.js
  1:9   error  setService is defined but never used  no-unused-vars
  2:2   error  Expected { after 'if' condition       curly
  3:14  error  "getParts" is not defined             no-undef

✖ 3 problems (3 errors, 0 warnings)
```

Here is my .eslintrc file:

```js
{
  "env": {
    "browser": true,
    "node": true,
    "es6": true
  },
  "ecmaFeatures": {
    "modules": true
  },
  "rules": {
    "quotes": [2, "single"],
    "no-underscore-dangle": 0
  }
}
```

Which is to say: I don't think it's a bug :)

**Comment by nzakas** [comment]
@tracker1 can you confirm the version of ESLint you're using? And are you using b as babel-eslint?

**Comment by gyandeeps** [comment]
I am going to close this issue as @xjamundx was to able to reproduce this issue.
@tracker1 Feel free to reopen this issue if you think this issue still exists.
